### ISSUE TITLE
micro service crashed with panic: runtime error: integer divide by zero

### ISSUE THREAD

**Issue by xmlking** [micro/go-micro#868]
Got panic when I am testing  following scenario  on my laptop. 
I am trying to see if `selector=static` and `http broker` works...  and it needed it worked, but service  crashed after serving request .... 

 account-srv -- gRPC- -> greeter-srv
 account-srv -- publish --> emailersrv-srv

### Env 
	github.com/micro/go-micro v1.13.0

Setup my macbook with multiple IPs and hostnames as below, so that I can run multiple services on same `8080` port

Add alias IPs for lo0
```bash
ifconfig lo0 alias 127.0.0.3
ifconfig lo0 alias 127.0.0.2
```
add local dns by editing  `/etc/hosts`
```
127.0.0.1 accountsrv
127.0.0.2 emailersrv
127.0.0.3 greetersrv
```
Run three services... 
```bash
go run ./srv/account   --selector=static --server_address=accountsrv:8080
go run ./srv/emailer   --selector=static --server_address=emailersrv:8080
go run ./srv/greeter   --selector=static --server_address=greetersrv:8080
```

Call with  micro cli
```
micro --selector=static  call  accountsrv UserService.Create '{"username": "sumo", "firstName": "sumo", "lastName": "demo", "email": "sumo@demo.com"}'
```
This trigger execution of this handler method:
https://github.com/xmlking/micro-starter-kit/blob/develop/srv/account/handler/user.handler.go#L108

#### `accountsrv` logs with error 
```
go run ./srv/account --selector=static --server_address=accountsrv:8080                  
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T16:24:02-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T16:24:02-07:00] loading configuration from file: config/config.yaml 
DEBU[2019-10-18T16:24:02-07:00] Client type: grpc or regular? *micro.clientWrapper 
DEBU[2019-10-18T16:24:02-07:00] CREATE TABLE "users" ("created_at" datetime NOT NULL,"deleted_at" datetime,"email" varchar(255) NOT NULL,"first_name" varchar(255) NOT NULL,"id" uuid,"last_name" varchar(255) NOT NULL,"updated_at" datetime NOT NULL,"username" varchar(100) NOT NULL , PRIMARY KEY ("id")) [196.572µs]  module=gorm
DEBU[2019-10-18T16:24:02-07:00] CREATE INDEX idx_users_deleted_at ON "users"(deleted_at)  [75.21µs]  module=gorm
DEBU[2019-10-18T16:24:02-07:00] CREATE TABLE "profiles" ("avatar" varchar(255),"birthday" datetime NOT NULL,"created_at" datetime NOT NULL,"deleted_at" datetime,"gender" varchar(255),"id" uuid,"tz" varchar(255),"updated_at" datetime NOT NULL,"user_id" varchar(255) , PRIMARY KEY ("id")) [83.01µs]  module=gorm
DEBU[2019-10-18T16:24:02-07:00] CREATE INDEX idx_profiles_deleted_at ON "profiles"(deleted_at)  [66.285µs]  module=gorm
INFO[2019-10-18T16:24:02-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T16:24:02-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T16:24:02-07:00] Server [grpc] Listening on 127.0.0.1:8080    
INFO[2019-10-18T16:24:02-07:00] Broker [http] Connected to [::]:64628        
INFO[2019-10-18T16:24:02-07:00] Registering node: accountsrv-d550db95-9a7a-4054-9731-82404aecfb7f 
DEBU[2019-10-18T16:25:06-07:00] Server-side request                           category=LogWrapper ctx="map[:authority:accountsrv:8080 Remote:127.0.0.1:64733 content-type:application/grpc+json user-agent:grpc-go/1.24.0]" method=UserService.Create service=accountsrv
INFO[2019-10-18T16:25:06-07:00] Received UserHandler.Create request          
INFO[2019-10-18T16:25:06-07:00] Received userRepository.Exist request {<nil> <nil> sumo@demo.com sumo  demo <nil> <nil> sumo} 
DEBU[2019-10-18T16:25:06-07:00] SELECT count(*) FROM "users"  WHERE "users"."deleted_at" IS NULL AND ((username = 'sumo')) [110.914µs]  module=gorm
DEBU[2019-10-18T16:25:06-07:00] SELECT count(*) FROM "users"  WHERE "users"."deleted_at" IS NULL AND ((email = 'sumo@demo.com')) [40.897µs]  module=gorm
DEBU[2019-10-18T16:25:06-07:00] INSERT  INTO "users" ("created_at","deleted_at","email","first_name","id","last_name","updated_at","username") VALUES ('2019-10-18 16:25:06',NULL,'sumo@demo.com','sumo','4cd71650-c2f3-4254-9073-c41fe8800229','demo','2019-10-18 16:25:06','sumo') [80.912µs]  module=gorm
INFO[2019-10-18T16:25:06-07:00] Got greeterService responce Hello sumo       
panic: runtime error: integer divide by zero

goroutine 32 [running]:
github.com/micro/go-micro/broker.(*httpBroker).Publish.func2(0xc00036ef70, 0x2, 0x2, 0xc000112240, 0x109, 0x120)
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:603 +0x69f
github.com/micro/go-micro/broker.(*httpBroker).Publish.func3(0xc0001620e0, 0xc0000ec5a0, 0xa, 0xc0000ccf30, 0xc00036ef70, 0x2, 0x2)
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:623 +0xd2
created by github.com/micro/go-micro/broker.(*httpBroker).Publish
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:615 +0x403
exit status 2

```

#### emailer logs
```
go run ./srv/emailer --selector=static --server_address=emailersrv:8080   
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T16:24:34-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T16:24:34-07:00] loading configuration from file: config/config.yaml 
INFO[2019-10-18T16:24:34-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T16:24:34-07:00] Server [grpc] Listening on 127.0.0.2:8080    
INFO[2019-10-18T16:24:34-07:00] Broker [http] Connected to [::]:64678        
INFO[2019-10-18T16:24:34-07:00] Registering node: emailersrv-5e9edfca-9cda-49c4-8769-ded47cfd274c 
INFO[2019-10-18T16:25:06-07:00] EmailSubscriber: Received msg to:"sumo@demo.com"  with metadata map[:authority:accountsrv:8080 Micro-From-Service:accountsrv Remote:127.0.0.1:64733 content-type:application/grpc+json user-agent:grpc-go/1.24.0] 
INFO[2019-10-18T16:25:06-07:00] in Welcome                                   
INFO[2019-10-18T16:25:06-07:00] in SendEmail.Send                            
DEBU[2019-10-18T16:25:06-07:00] sending email to: [sumo@demo.com] from: xmlking@gmail.com, subject: Welcome, body: From: xmlking@gmail.com<br />
    To: sumo@demo.com<br />
    Subject: Welcome<br />
    MIME-version: 1.0<br />
    Content-Type: text/html; charset=&quot;UTF-8&quot;<br />
    <br />
    Hi ! 
INFO[2019-10-18T16:25:07-07:00] Function: Received msg to:"sumo@demo.com"  with metadata map[:authority:accountsrv:8080 Micro-From-Service:accountsrv Remote:127.0.0.1:64733 content-type:application/grpc+json user-agent:grpc-go/1.24.0]
```

#### greeter logs
```
go run ./srv/greeter --selector=static --server_address=greetersrv:8080   
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T16:24:54-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T16:24:54-07:00] loading configuration from file: config/config.yaml 
INFO[2019-10-18T16:24:54-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T16:24:54-07:00] Server [grpc] Listening on 127.0.0.3:8080    
INFO[2019-10-18T16:24:54-07:00] Broker [http] Connected to [::]:64712        
INFO[2019-10-18T16:24:54-07:00] Registering node: greetersrv-a918ad1d-3b14-43cb-ac28-c048bc407021 
DEBU[2019-10-18T16:25:06-07:00] Server-side request                           category=LogWrapper ctx="map[:authority:greetersrv:8080 Remote:127.0.0.3:64735 content-type:application/grpc+proto micro-from-service:accountsrv user-agent:grpc-go/1.24.0 x-from-id:script x-user-id:john]" method=Greeter.Hello service=greetersrv
INFO[2019-10-18T16:25:06-07:00] Received Greeter.Hello request
```

**Comment by xmlking** [comment]
If I add `--registry=memory`  to `account-srv`, is not crashing this time but it is not publishing message to `emailer-srv`
```
go run ./srv/account --selector=static  --server_address=accountsrv:8080 --registry=memory
```

**Comment by xmlking** [comment]
default setup also crashing... 
``` 
go run ./srv/account 
go run ./srv/emailer 
go run ./srv/greeter 
```

```
micro list services
accountsrv
emailersrv
greetersrv
```

```
micro call accountsrv UserService.Create '{"username": "sumo", "firstName": "sumo", "lastName": "demo", "email": "sumo@demo.com"}'
```

### accountsrv logs 

```
go run ./srv/account
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T19:36:05-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T19:36:05-07:00] loading configuration from file: config/config.yaml 
DEBU[2019-10-18T19:36:05-07:00] Client type: grpc or regular? *micro.clientWrapper 
DEBU[2019-10-18T19:36:05-07:00] CREATE TABLE "users" ("created_at" datetime NOT NULL,"deleted_at" datetime,"email" varchar(255) NOT NULL,"first_name" varchar(255) NOT NULL,"id" uuid,"last_name" varchar(255) NOT NULL,"updated_at" datetime NOT NULL,"username" varchar(100) NOT NULL , PRIMARY KEY ("id")) [210.018µs]  module=gorm
DEBU[2019-10-18T19:36:05-07:00] CREATE INDEX idx_users_deleted_at ON "users"(deleted_at)  [84.222µs]  module=gorm
DEBU[2019-10-18T19:36:05-07:00] CREATE TABLE "profiles" ("avatar" varchar(255),"birthday" datetime NOT NULL,"created_at" datetime NOT NULL,"deleted_at" datetime,"gender" varchar(255),"id" uuid,"tz" varchar(255),"updated_at" datetime NOT NULL,"user_id" varchar(255) , PRIMARY KEY ("id")) [83.984µs]  module=gorm
DEBU[2019-10-18T19:36:05-07:00] CREATE INDEX idx_profiles_deleted_at ON "profiles"(deleted_at)  [64.324µs]  module=gorm
INFO[2019-10-18T19:36:05-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T19:36:05-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T19:36:05-07:00] Server [grpc] Listening on [::]:59718        
INFO[2019-10-18T19:36:05-07:00] Broker [http] Connected to [::]:59719        
INFO[2019-10-18T19:36:05-07:00] Registering node: accountsrv-246d5d29-1182-4de9-97a6-abbf6c5562e4 
DEBU[2019-10-18T19:36:20-07:00] Server-side request                           category=LogWrapper ctx="map[:authority:192.168.2.242:59718 Remote:192.168.2.242:59725 content-type:application/grpc+json user-agent:grpc-go/1.24.0]" method=UserService.Create service=accountsrv
INFO[2019-10-18T19:36:20-07:00] Received UserHandler.Create request          
INFO[2019-10-18T19:36:20-07:00] Received userRepository.Exist request {<nil> <nil> sumo@demo.com sumo  demo <nil> <nil> sumo} 
DEBU[2019-10-18T19:36:20-07:00] SELECT count(*) FROM "users"  WHERE "users"."deleted_at" IS NULL AND ((username = 'sumo')) [133.022µs]  module=gorm
DEBU[2019-10-18T19:36:20-07:00] SELECT count(*) FROM "users"  WHERE "users"."deleted_at" IS NULL AND ((email = 'sumo@demo.com')) [50.241µs]  module=gorm
DEBU[2019-10-18T19:36:20-07:00] INSERT  INTO "users" ("created_at","deleted_at","email","first_name","id","last_name","updated_at","username") VALUES ('2019-10-18 19:36:20',NULL,'sumo@demo.com','sumo','d0f010ef-d72d-4f46-84aa-34fc1afb36e6','demo','2019-10-18 19:36:20','sumo') [89.36µs]  module=gorm
panic: runtime error: integer divide by zero

goroutine 105 [running]:
github.com/micro/go-micro/broker.(*httpBroker).Publish.func2(0xc0003c6090, 0x2, 0x2, 0xc000116240, 0x111, 0x120)
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:603 +0x69f
github.com/micro/go-micro/broker.(*httpBroker).Publish.func3(0xc0003b8000, 0xc00044c2d0, 0xa, 0xc000393080, 0xc0003c6090, 0x2, 0x2)
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:623 +0xd2
created by github.com/micro/go-micro/broker.(*httpBroker).Publish
        /Users/sumo/go/pkg/mod/github.com/micro/go-micro@v1.13.0/broker/http_broker.go:615 +0x403
exit status 2
```

### emailersrv logs
```
go run ./srv/emailer                                 
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T19:35:14-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T19:35:14-07:00] loading configuration from file: config/config.yaml 
INFO[2019-10-18T19:35:14-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T19:35:14-07:00] Server [grpc] Listening on [::]:59702        
INFO[2019-10-18T19:35:14-07:00] Broker [http] Connected to [::]:59703        
INFO[2019-10-18T19:35:14-07:00] Registering node: emailersrv-7499cd5e-7f70-4977-ba85-19557ba33071
```

###  greetersrv logs
```
go run ./srv/greeter                                 
INFO[0000] loading configuration from file: config/config.yaml 
INFO[2019-10-18T19:35:22-07:00] Logger set to Logrus with:                    format=text logLevel=debug
INFO[2019-10-18T19:35:22-07:00] loading configuration from file: config/config.yaml 
INFO[2019-10-18T19:35:22-07:00] version     : untouched
build date  : 
go version  : go1.13.1
go compiler : gc
platform    : darwin/amd64
git commit  : 
git branch  : 
git state   : 
git summary :  
INFO[2019-10-18T19:35:22-07:00] Server [grpc] Listening on [::]:59706        
INFO[2019-10-18T19:35:22-07:00] Broker [http] Connected to [::]:59707        
INFO[2019-10-18T19:35:22-07:00] Registering node: greetersrv-12a06adc-4b8a-49e7-a4f7-beb89967f336
```

**Comment by milosgajdos** [comment]
This should be fixed in `1.13.1` by https://github.com/micro/go-micro/commit/a89d1edc419f54e568633d903a4ad7642daf6bf2
